cmake/SeastarDependencies.cmake

```diff
@@ -159,11 +159,11 @@ macro (seastar_find_dependencies)
   # ProtobufConfig.cmake provided by protobuf defines this linkage. so we try
   # the CMake package configuration file first, and fall back to CMake's
   # FindProtobuf module.
-  find_package (Protobuf 2.5.0
+  find_package (Protobuf 22.0...25.3
```

Comment:
I don't understand how 2.5.0 worked before and now it's 22.0..22.3. I understood there are two versioning systems, but you don't change versioning systems here.

Reply:
Also, does 26 not work?

Reply:
@avikivity 

> I don't understand how 2.5.0 worked before and now it's 22.0..22.3. I understood there are two versioning systems, but you don't change versioning systems here.

22.0 is the first version which uses abseil. so i wanted to make this explicit. and i didn't want to mix two versioning systems in a single condition, like `find_package (Protobuf 2.5...25.3)` even if it would yield the expected outcome. but it's just weird..

> Also, does 26 not work?

yeah, i asked myself when writing these numbers. but i wanted to avoid adding versions in future and wanted to avoid pretending that we support versions we haven't even compiled with. that's why it looks this way. but it didn't pan out. 

anyway, i am pushing an updated version.

Reply:
What I didn't understand is the version jump from 2 to 22. Was the numbering changed? Or do the version numbers refer to different things?

Reply:
please note, in the latest revision i stopped using 22 as the version number. and sorry, i failed to explain my reasoning on the previous revision in an understandable way. let me rephrase it.

more like it refer to a different thing. 

actually i didn't bumped the version from 2 to 22. it was because, the CMake configuration file shipped by Protobuf uses the library's version, i.e., 22 and 23..., while the `FindProtobuf` module uses the language binding's version, like 2.5, 3.12, 3.21, 4.22, ....  .  the cutting edge distros are more likely to use the CMake configuration file shipped by Protobuf, so i intended to use the library's version when finding the package using the config mode, and i need to pick a library version . instead of trying to map 2.5 to "protobuf v5.0" which does not exst at all. i guess the reason is that it predates the version when protobuf developers decided to using this new versioning approach. so, i used v22, which is the first [version](https://github.com/protocolbuffers/protobuf/releases/tag/v22.0) which uses abseil. so that i can assure that the change works.

Reply:
@avikivity ^

Reply:
ah, thanks.